src/adapter/src/coord.rs

```diff
@@ -833,6 +833,11 @@ impl<S: Append + 'static> Coordinator<S> {
                         }
                     }
 
+                    let source_status_collection_id =
```

Comment:
Moving this outside of the for loop would save some time since we'll always get the same ID.

Reply:
Nice catch, thanks!